integrations/azure-devops/azure_devops/webhooks/webhook_event_observer.py

```diff
@@ -0,0 +1,30 @@
+import asyncio
+from collections import defaultdict
+from typing import Any, Awaitable, Callable
+from .webhook_event import WebhookEvent
+
+Observer = Callable[[dict[str, Any]], Awaitable[Any]]
+
+
+class WebhookEventObserver:
+    def __init__(self) -> None:
+        self._observers: dict[str, list[Observer]] = defaultdict(list)
+
+    def on(self, events: list[WebhookEvent], observer: Observer) -> None:
+        for event in events:
+            self._observers[self._get_observer_key_from_webhook_event(event)].append(
+                observer
+            )
+
+    async def notify(self, event: WebhookEvent, body: dict[str, Any]) -> Awaitable[Any]:
+        return asyncio.gather(
+            *(
+                observer(body)
+                for observer in self._observers.get(
+                    self._get_observer_key_from_webhook_event(event), []
+                )
+            )
+        )
```

Comment:
handle any exceptions
you can pass parameter to this function to make it return the exceptions as a result instead of raising